packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx

```diff
@@ -19,25 +19,35 @@ import {
 } from '@mui/material'
 import { parse, ParseError } from 'papaparse'
 import { RJSFSchema } from '@rjsf/utils'
-type JSONArrayEditorProps = {
-  value: string[]
+import TextWidget from '../SchemaDrivenAnnotationEditor/widget/TextWidget'
+import { DateTimeWidget } from '../SchemaDrivenAnnotationEditor/widget/DateTimeWidget'
+import { SelectWidget } from '../SchemaDrivenAnnotationEditor/widget/SelectWidget'
+import { BooleanWidget } from '../SchemaDrivenAnnotationEditor/widget/BooleanWidget'
+import { getTransformErrors } from '../SchemaDrivenAnnotationEditor/AnnotationEditorUtils'
+import ErrorListTemplate from '../SchemaDrivenAnnotationEditor/template/ErrorListTemplate'
+
+export type JSONArrayEditorProps = {
+  value?: string[]
+  arrayItemDefinition?: JSONSchema7Definition
   onChange: (newValue: string[]) => void
   onSubmit: (formData: string[]) => void
 }
 
-const arraySchema: JSONSchema7 = {
-  $schema: 'http://json-schema.org/draft-07/schema#',
-  type: 'array',
-  items: {
-    type: 'string',
-  },
+function getSchema(
+  definition: JSONSchema7Definition = { type: 'string' },
+): JSONSchema7 {
+  return {
+    $schema: 'http://json-schema.org/draft-07/schema#',
+    type: 'array',
+    items: definition,
+  }
 }
```

Comment:
allow customizing the items definition, so this could be an array of dates, integers, etc.